### PR TITLE
24 domas  update aim trainer game

### PR DESCRIPTION
**Pull request by Dax0s**
- add LINQ
- add extension methods
- fix a bug where if you go back when a game is running the dots keep spawning
-  switch to var
-  move out and change type of consts
-  convert Vector2 constructor
-  move to vite
-  make url a const

### CODE REVIEW

File: frontend/src/games/aim-trainer-game.tsx [hunk 1]
```diff
@@ -50,6 +53,7 @@ async function fetchGameStartInfo(
     });
     return (await tmp.json()) as GameStartResponse;
   } catch (e) {
+    console.log(e);
```
Comment (Aidas21J) [inline frontend/src/games/aim-trainer-game.tsx]: Cia probably labiau tiktu ```console.error(e)``` arba throwint ```new Error(e)```.

File: frontend/src/games/aim-trainer-game.tsx [hunk 2]
```diff
@@ -60,23 +64,30 @@ function styleElement(
...
+  const dots: MutableRefObject<Array<HTMLDivElement>> = useRef([]);
+
   function removeElement(element: HTMLDivElement) {
     element.remove();
     setDotsLeft((prevDots) => prevDots - 1);
   }
 
+  function removeAllDots() {
+    dots.current.forEach(removeElement);
+
```
Comment (Aidas21J) [inline frontend/src/games/aim-trainer-game.tsx]: Nzn ar cia reikia tuscios eilutes. Jei kazka pries mergindamas dar keisi, consider removing it, jei ne, tai whatever nbd.

File: frontend/src/utils/consts.ts
Comment (Aidas21J) [inline frontend/src/utils/consts.ts:1]: Siaip ner blogai, bet, personally, ```constants.ts``` netrumpinciau i ```consts.ts```, nes tiesiog nlb matau, kodel tai apsimoketu daryt.

**Review by Aidas21J — APPROVED** [review]
Very good. Backendas ne iki galo suformatuotas, bet ten yra biski niuansu, tai realiai gali palikt as arba ainius su savo pr viska sutvarkysim.

File: frontend/src/utils/consts.ts
Comment (Dax0s) [reply frontend/src/utils/consts.ts:1]: as taip padariau kadangi tsg javascript konstantos yra kuriamos su const keyword, bet yeah, pervadinsiu i constants

File: frontend/src/games/aim-trainer-game.tsx (hunk 2 above)
Comment (Dax0s) [reply frontend/src/games/aim-trainer-game.tsx]: Okey, paremovinsiu

File: frontend/src/games/dot-count-game.tsx [hunk 3]
```diff
@@ -42,7 +43,7 @@ const DotCountGame = () => {
   const fetchInfo = async (maxDots: number) => {
     try {
       const response = await fetch(
-        `http://localhost:5252/DotCountGame?maxDots=${maxDots}`,
+        `${BACKEND_URL}/DotCountGame?maxDots=${maxDots}`,
       );
       const canvasInfo = (await response.json()) as DotCanvasInfo;
       console.log(canvasInfo);
```
Comment (AtomicTacoDev) [inline frontend/src/games/dot-count-game.tsx]: ar bereikia sito printo? Mane bent triggerina kai console uzkista nereikalinga info:D

File: frontend/src/games/reaction-game.tsx [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import { useRef, useState } from 'react';
 import { delay } from '@/utils/timing';
+import { BACKEND_URL } from '@/utils/consts';
 
 async function fetchWaitTime() {
-  const tmp = await fetch('http://localhost:5252/reactiontime');
+  const tmp = await fetch(`${BACKEND_URL}/localhost:5252/reactiontime`);
```
Comment (AtomicTacoDev) [inline frontend/src/games/reaction-game.tsx]: tik http:// dali pakeitei

File: frontend/src/games/aim-trainer-game.tsx [hunk 5]
```diff
@@ -3,7 +3,7 @@ import Difficulty from '@/components/aim-trainer-game/Difficulty';
 import DifficultyPicker from '@/components/aim-trainer-game/DifficultyPicker';
 import StartGameButton from '@/components/aim-trainer-game/StartGameButton';
 import { delay } from '@/utils/timing';
-import { BACKEND_URL } from '@/utils/consts';
+import { BACKEND_URL } from '@/utils/contants';
```
Comment (AtomicTacoDev) [inline frontend/src/games/aim-trainer-game.tsx]: visur tas pats typo, praleidai "conStants"

**Review by AtomicTacoDev — APPROVED** [review]

**Pull request merged**